tests/integration/solidity_contracts/PlainOpcodes/PlainOpcodes.sol

```diff
@@ -127,4 +128,82 @@ contract PlainOpcodes {
             extcodecopy(target, add(extcode, 0x20), offset, size)
         }
     }
+
+    function testShouldAlwaysRevert() public pure {
+        revert("This always reverts");
+    }
+
+    function testCallingContextShouldPropogateRevertFromSubContext() public {
+        Parent parent = new Parent();
+        parent.triggerRevert();
+
+    }
+
+    function testCallingContextShouldPropogateRevertFromSubContextOnCreate() public {
+        ContractRevertsOnConstruction doomedContract = new ContractRevertsOnConstruction();
+
+    }    
+    
+    function testShouldRevertViaCall() external returns (address address_, bool triggerRevert ) {
+        Parent parent = new Parent();
+
+        (bool success, ) = address(parent).call(abi.encodeWithSignature("triggerRevert()"));
```

Comment:
Does the `returnData` return the revert reason in case of revert?
You could then add here your 

```
assert(address(parent.child()) == address(0)); 
```

Reply:
I pushed in PlainOpcodes.t.sol what it would look like to assert the error message in soliditiy. trying to run the same solidity code in kkrt resulted in a much longer test time (i never bothered waiting for it to finish once it got to the 5x time difference) so I opted to just return the returndata and test for the revert in python